tests/clims/models/test_substance.py

```diff
@@ -211,6 +211,29 @@ def do_asserts(child):
         do_asserts(child)
 
 
+class TestSubstanceType(TestCase):
+    def setUp(self):
+        pass
+
+    def test_it(self):
+        # TODO: This is really an extensible test
+        plugin = testutils.create_plugin()
+        extensibles.register(plugin, GemstoneSample)
```

Comment:
It would be nice with some documentation on how plugins are to be used in commonlims. (maybe there already are?). It seems to me that one single database may contain multiple plugins, and each extensible type is associated with a certain plugin? How does this relate to e.g. stockholm and snpseq working with the same lims?

Reply:
Yes. There is some documentation already and I think we should certainly document it more when we have less of a moving target.

The idea is that one can have 0..n plugins running and even different plugins depending on the organization. The types are as you say associated with a plugin.

So we will be able to have e.g. one sample object as required by lab1 and another by lab2. They can be as much alike the other or differ as much. They can also inherit from another if required. In the end they are just key/value pairs in the `Extensible` system.